independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/ArcTestClassLoader.java

```diff
@@ -0,0 +1,39 @@
+package io.quarkus.arc.test;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.Objects;
+
+import io.quarkus.arc.ComponentsProvider;
+import io.quarkus.arc.ResourceReferenceProvider;
+
+public class ArcTestClassLoader extends URLClassLoader {
```

Comment:
Unrelated: I looked at this recently and thought, does this need to be `URLClassLoader`? We never pass any `URL` to it. Extending just `ClassLoader` here should be enough IMHO.

Reply:
Ok, why not. I don't see a difference though...